# Исследование надежности заемщиков


## Откройте таблицу и изучите общую информацию о данных



**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

`/datasets/data.csv`

In [1]:
import pandas as pd


In [2]:
try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [6]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [7]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [8]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [9]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [10]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [11]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [12]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [13]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [14]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [15]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [16]:
print(data.duplicated().sum()) 

71


In [17]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [18]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [20]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

In [23]:
data

children  days_employed  dob_years education  education_id  \
0             1    8437.673028         42    высшее             0   
1             1    4024.803754         36   среднее             1   
2             0    5623.422610         33   среднее             1   
3             3    4124.747207         32   среднее             1   
4             0  340266.072047         53   среднее             1   
...         ...            ...        ...       ...           ...   
21520         1    4529.316663         43   среднее             1   
21521         0  343937.404131         67   среднее             1   
21522         1    2113.346888         38   среднее             1   
21523         3    3112.481705         38   среднее             1   
21524         2    1984.507589         40   среднее             1   

          family_status  family_status_id gender income_type  debt  \
0       женат / замужем                 0      F   сотрудник     0   
1       женат / замужем                 0      F   сотрудник     0   
2       женат / замужем                 0      M   сотрудник     0   
3       женат / замужем                 0      M   сотрудник     0   
4      гражданский брак                 1      F   пенсионер     0   
...                 ...               ...    ...         ...   ...   
21520  гражданский брак                 1      F   компаньон     0   
21521   женат / замужем                 0      F   пенсионер     0   
21522  гражданский брак                 1      M   сотрудник     1   
21523   женат / замужем                 0      M   сотрудник     1   
21524   женат / замужем                 0      F   сотрудник     0   

       total_income                       purpose total_income_category  \
0            253875                 покупка жилья                     B   
1            112080       приобретение автомобиля                     C   
2            145885                 покупка жилья                     C   
3            267628    дополнительное образование                     B   
4            158616               сыграть свадьбу                     C   
...             ...                           ...                   ...   
21520        224791             операции с жильем                     B   
21521        155999          сделка с автомобилем                     C   
21522         89672                  недвижимость                     C   
21523        244093  на покупку своего автомобиля                     B   
21524         82047         на покупку автомобиля                     C   

               purpose_category  
0      операции с недвижимостью  
1        операции с автомобилем  
2      операции с недвижимостью  
3         получение образования  
4            проведение свадьбы  
...                         ...  
21520  операции с недвижимостью  
21521    операции с автомобилем  
21522  операции с недвижимостью  
21523    операции с автомобилем  
21524    операции с автомобилем  

[21331 rows x 14 columns]

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Для целей анализа влияния разных вводных данных на возврат кредита в срок, создадим функцию, которая на вход принимает столбец, по которому мы анализируем данные, в теле функции метод groupby() принимает этот столбец, группируя данные, а методу agg() передаем название столбца, к которому применяем функции (sum и count). Далее, на основании полученных сгруппированных данных, функция создает новый столбец 'overdue_debt', значаение которого и есть процент невозврата кредита в срок по категориям. И последняя операция - отсортируем новую таблицу по столбцу 'overdue_debt' в порядке убывания.

In [24]:
def relation(category):
     data_grouped = data.groupby(category).agg({'debt':['sum','count']}) #группируем данные по нужному столбцу, ключом к методу agg выбираем столбец 'debt' и применяем сразу две агрегированные функции sum и count
     data_grouped['overdue_debt'] = data_grouped['debt']['sum'] / data_grouped['debt']['count'] #создаем новый столбец 'overdue_debt' в data_grouped, его значения - это процент невозврата кредита в срок по категориям
     data_grouped = data_grouped.sort_values(by = 'overdue_debt', ascending = False) #отсортируем наши данные по столбцу 'overdue_debt' в порядке убывания
     return data_grouped
relation(['children']) 

debt        overdue_debt
           sum  count             
children                          
4            4     41     0.097561
2          194   2052     0.094542
1          444   4808     0.092346
3           27    330     0.081818
0         1063  14091     0.075438
5            0      9     0.000000

**Вывод:** 
Из полученной таблицы мы можем заметить, что чем больше у заемщика детей, тем реже он берет кредиты, а вот оценить
как многодетность влияет на возврат кредита вовремя по нашим данным нам не удастся, потому как заемщик с 4-мя детьми 
получется самый ненадежный, а с 5-тью с точностью наоборот. Скорее всего у нас маленькая выборка. А вот заемщики у 
которых вовсе нет детей получаются самыми надежными. 

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Ответить на этот вопрос нам поможет ранее созданная функция, но на ее вход теперь направим столбец 'family_status'.

In [25]:
relation(['family_status'])

debt        overdue_debt
                       sum  count             
family_status                                 
Не женат / не замужем  273   2796     0.097639
гражданский брак       385   4134     0.093130
женат / замужем        927  12261     0.075606
в разводе               84   1189     0.070648
вдовец / вдова          63    951     0.066246

**Вывод:** 
На основе нашего анализа, можем сделать вывод, что люди не состоящие и не состоявшие в зарегестрированном браке чаще остальных
не возвращают кредиты в срок. А те, кто состоит или когда-либо состоял в зарегистрированном браке, более надежные заемщики.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

На этапе предобработки данных мы создали новый столбец датафрейма с категориями дохода заемщиков. Этот столбец нам понадобится
для ответа на вопрос выше. На вход нашей функции теперь пойдет столбец 'total_income_category'.

In [26]:
relation(['total_income_category'])

debt        overdue_debt
                        sum  count             
total_income_category                          
E                         2     22     0.090909
C                      1353  15921     0.084982
A                         2     25     0.080000
B                       354   5014     0.070602
D                        21    349     0.060172

**Вывод:** 

Можно отметить только, что люди с самым высоким доходом и так же с самым низким, реже остальных берут кредиты.В данной таблице группы  крайне несбаланисораны, поэтому мы будем учитывать только две самые многочисленные из них, а именно группу С(с доходом 50001–200000) и В (с доходом 200001–1000000), так как крайние категории по доходу не обладают достаточным объёмом клиентов для объективной оценки процента должников. Учитывая эти данные, можно сказать, что люди с доходом из группы В (200001–1000000) меньше нарушают свои кредитные обязательства, чем люди с более низким доходом из группы С.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Для анализа влияния цели кредита на возврат его в срок используем столбец с категориями целей кредита, 
который мы и отправим на вход нашей функции.

In [27]:
relation(['purpose_category'])

debt        overdue_debt
                          sum  count             
purpose_category                                 
операции с автомобилем    400   4279     0.093480
получение образования     369   3988     0.092528
проведение свадьбы        183   2313     0.079118
операции с недвижимостью  780  10751     0.072551

**Вывод:** 

Самые надежные заемщики берут кредиты на операции с недвижимостью, оно и понятно, шутки с недвижимостью плохи:)
Кредиты на операции с автомобилем более подвержены риску невозврата в срок, может это связано как-то с тем, что
автомобили могут и в аварию попасть и сломаться, в общем - дело рисковое.  

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* В процессе предобработки данных мы выяснили, что в нашем датафрейме пропуски встречались в столбцах 'days_employed' 
 и 'total_income'. Возможно появление этих пропусков вызвал человеческий фактор,но так как количество пропусков одинаковое в двух столбцах (2174), возможно это означает, что эти заемщики никогда не работали и не получали дохода. Так же возможной причиной пропусков данных может быть умышленное сокрытие информации, например, для того чтобы она не повлияла на решение о выдаче кредита.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Для заполнения пропусков количественных переменных можно применить среднее значение или медианное. Среднее значение более подходит для последовательных значений с одинаковым шагом, и некорректно характеризует данные, когда некоторые значения сильно выделяются среди большинства. Поэтому медианное значение не ведет нас в заблуждение, если данные сильно разнятся. 

### Шаг 4: общий вывод.


Целью моего исследования было определить влияние на погашение кредита в срок различных вводных данных, а именно: количества детей у заемщика('children'), семейного положения ('family_status'), уровня дохода ('total_income') и целей , на которые берется кредит ('purpose') .
 
 
 На этапе предобработки данных статистики о платёжеспособности клиентов, полученной от банка, я столкнулась с такими проблемами как: 


* Пропуск данных был отмечен в двух столбцах данных - общий трудовой стаж в днях ('days_employed')  и ежемесячный доход ('total_income'). Так как на ежемесячный доход сильнее всего влияет тип занятости, то пропуски в данных о ежемесячном доходе я заполнила медианным значением доходов по каждому типу из столбца - тип занятости ('income_type'). Пропуски в столбце - общий трудовой стаж в днях ('days_employed') заменила медианными значениями по каждому типу занятости ('income_type').

* В данных я встретила артефакты (аномалии) - отрицательное количество дней трудового стажа в столбце - общий трудовой стаж в днях ('days_employed'). Я заменила все отрицательные значения положительными. Так же у двух типов занятости (безработные и пенсионеры) я обнаружила аномально большие значения трудового стажа, но исправлять их не посчитала целесообразным. В столбце - количество детей в семье ('children') было два аномальных значения, я удалила строки, в которых встречаются такие аномальные значения из датафрейма.
* Заменила тип данных и удалила неявные дубликаты там, где это было необходимо.
* Провела категоризацию данных о ежемесячном доходе, выделила категории и создала новый столбец с категориями дохода ('total_income_category'): 
0–30000 — 'E';
30001–50000 — 'D';
50001–200000 — 'C';
200001–1000000 — 'B';
1000001 и выше — 'A'
* Провела категоризацию данных о целях кредита, выделила категории и создала новый столбец с категориями целей ('purpose_category'):
'операции с автомобилем',
'операции с недвижимостью',
'проведение свадьбы',
'получение образования'.

#    Результаты анализа данных:
#### 1. Есть ли зависимость между количеством детей и возвратом кредита в срок?

   Мои результаты анализа по вопросу зависимости возврата кредита в срок от количества детей заемщика основываются на выборке заемщиков с количеством детей от 0 до 2, которые занимают примерно 98 процентов от всей выборки. Анализ показал, что заемщики, у которых нет детей имеют наименьший процент невозврата кредита в срок - 0.075438, с 1 ребенком - 0.092346, с 2 детьми - 0.094542. Так же можно отметить, что, чем больше у заемщика детей, тем реже он берет кредиты.

####    2. Есть ли зависимость между семейным положением и возвратом кредита в срок?

   На основе проведенного анализа , могу сделать вывод, что люди не состоящие и не состоявшие в зарегестрированном браке, чаще остальных не возвращают кредиты в срок, процент невозврата по категориям: 
   Не женат / не замужем	273	2796	0.097639
   гражданский брак	385	4134	0.093130
   А те, кто состоит или когда-либо состоял в зарегистрированном браке, более надежные заемщики.
   женат / замужем	=	0.075606
   в разводе	=	0.070648
   вдовец / вдова	=	0.066246

####   3. Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

   Для ответа на этот вопрос я учитывала две самые многочисленные категории, а именно группу С(с доходом 50001–200000) и В (с доходом 200001–1000000), так как крайние категории по доходу не обладают достаточным объёмом клиентов для объективной оценки процента должников. Учитывая эти данные, можно сказать, что люди с доходом из группы В (200001–1000000)(процент невозврата в срок = 0.070602) меньше нарушают свои кредитные обязательства, чем люди с более низким доходом из группы С (50001–200000)(процент невозврата в срок = 0.084982).
   Так же можно отметить, что люди с самым высоким доходом и с самым низким, реже остальных берут кредиты.

####    4. Как разные цели кредита влияют на его возврат в срок?
   Самые надежные заемщики берут кредиты на операции с недвижимостью (процент невозврата в срок = 0.072551). Кредиты на операции с автомобилем наиболее подвержены риску невозврата в срок(процент невозврата в срок = 0.093480). Между ними расположились категории - получение образования (процент невозврата в срок =0.092528) и проведение свадьбы(процент невозврата в срок =0.079118).
 

   Сложив результаты анализа разных показателей можно сделать обощенный вывод о том, что самые надежные заемщики - те, которые состоят или ранее состояли в зарегестрированным браке и не имеют детей. Более надежный заемщик чаще обладает уровнем дохода  В (200001–1000000), а самая надежная цель кредита - операции с недвижимостью.


  Для дополнительного, более детального анализа одновременного влияния сразу всех анализируемых показателях клиентов на надежность их как заемщиков, я создала сводную таблицу, в которой видно, к примеру, что люди у которых нет детей, находящиеся в зарегистрированном браке, с уровнем дохода С (50001–200000), взявшие кредит на операции с недвижимостью - довольно-таки надежные заемщики с низким процентом невозврата кредита в срок (0.067451)

In [28]:
data_pivot = data.pivot_table(index = ['children','family_status','total_income_category','purpose_category'], values='debt', aggfunc=['sum', 'count']) #создали сводную таблицу с нашими столбцами, применили сразу две агрегирующие функции
#data_pivot.head(30)
data_pivot.columns = ['_'.join(col).strip() for col in data_pivot.columns.values] #избавились от мультииндексов в названиях столбцов
data_pivot = data_pivot.reset_index()
data_pivot['overdue_debt'] = data_pivot['sum_debt'] / data_pivot['count_debt'] #создаем новый столбец - процент невозврата кредита в срок
data_pivot_sort = data_pivot.sort_values(by = 'count_debt', ascending = False) # отсортируем таблицу по 'count_debt' по убыванию

data_pivot_sort.head(30)

children          family_status total_income_category  \
53          0        женат / замужем                     C   
108         1        женат / замужем                     C   
52          0        женат / замужем                     C   
54          0        женат / замужем                     C   
40          0       гражданский брак                     C   
50          0        женат / замужем                     B   
5           0  Не женат / не замужем                     C   
151         2        женат / замужем                     C   
38          0       гражданский брак                     C   
107         1        женат / замужем                     C   
109         1        женат / замужем                     C   
105         1        женат / замужем                     B   
96          1       гражданский брак                     C   
4           0  Не женат / не замужем                     C   
49          0        женат / замужем                     B   
25          0         вдовец / вдова                     C   
51          0        женат / замужем                     B   
36          0       гражданский брак                     B   
6           0  Не женат / не замужем                     C   
16          0              в разводе                     C   
2           0  Не женат / не замужем                     B   
152         2        женат / замужем                     C   
150         2        женат / замужем                     C   
148         2        женат / замужем                     B   
37          0       гражданский брак                     C   
39          0       гражданский брак                     C   
94          1       гражданский брак                     C   
65          1  Не женат / не замужем                     C   
104         1        женат / замужем                     B   
24          0         вдовец / вдова                     C   

             purpose_category  sum_debt  count_debt  overdue_debt  
53   операции с недвижимостью       212        3143      0.067451  
108  операции с недвижимостью        91        1255      0.072510  
52     операции с автомобилем        86        1231      0.069862  
54      получение образования        97        1182      0.082064  
40         проведение свадьбы        92        1153      0.079792  
50   операции с недвижимостью        62         999      0.062062  
5    операции с недвижимостью        71         955      0.074346  
151  операции с недвижимостью        58         633      0.091627  
38   операции с недвижимостью        42         491      0.085540  
107    операции с автомобилем        51         487      0.104723  
109     получение образования        47         450      0.104444  
105  операции с недвижимостью        30         417      0.071942  
96         проведение свадьбы        46         397      0.115869  
4      операции с автомобилем        49         396      0.123737  
49     операции с автомобилем        30         388      0.077320  
25   операции с недвижимостью        19         382      0.049738  
51      получение образования        20         362      0.055249  
36         проведение свадьбы        22         350      0.062857  
6       получение образования        38         337      0.112760  
16   операции с недвижимостью        22         314      0.070064  
2    операции с недвижимостью        24         313      0.076677  
152     получение образования        28         253      0.110672  
150    операции с автомобилем        25         234      0.106838  
148  операции с недвижимостью        13         226      0.057522  
37     операции с автомобилем        23         209      0.110048  
39      получение образования        29         195      0.148718  
94   операции с недвижимостью        26         192      0.135417  
65   операции с недвижимостью        21         189      0.111111  
104    операции с автомобилем        13         182      0.071429  
24     операции с автомобилем        17   